clients/roscpp/src/libros/io.cpp

```diff
@@ -44,9 +44,16 @@
 #else
   #include <cstring> // strerror
   #include <fcntl.h> // for non-blocking configuration
+  #include <sys/epoll.h>
```

Comment:
This is going to cause trouble on POSIX platforms where epoll isn't a thing, especially OS X and BSD. I think you'll need to test for the header in CMake and then have an #ifdef that only optionally includes it.

You can do this easily with CMake's `check_include_file` macro, see: https://cmake.org/cmake/help/v3.5/module/CheckIncludeFile.html